### DISCUSSION TITLE
How to filter metrics by type

### DISCUSSION THREAD

**Discussion by Quyzi** [vectordotdev/vector#12969] [Q&A] [answered]
In my metrics environment, histograms and counters/gauges need to be handled differently. I'm trying to set up a filter to Separate out the histograms, but I can't seem to figure out how to match the type. 

I sent the internal metrics to a stdout sink as json, and got this event. 
```
{
  "name": "internal_metrics_cardinality_total",
  "namespace": "vector-agent",
  "tags":
    {
      "az": "local",
      "clusterGroupName": "local",
      "clusterName": "local",
      "project_id": "undefined",
      "region": "local",
      "serviceName": "service-agent",
    },
  "timestamp": "2022-06-03T18:19:35.076353386Z",
  "kind": "absolute",
  "counter": { "value": 263.0 },
}
```

I added this transform to filter out histogram metrics, 
```
  "agent_metrics_filter":
    "condition": "exists(.counter) || exists(.gauge)"
    "inputs":
      - "agent_metrics"
    "type": "filter"
```
But I get no output from it. According to the schema, there should be a `.counter` or `.gauge` field on metrics events. Am I approaching this wrong?

**Comment by spencergilbert** [comment, marked as answer]
🤔 I'm trying to remember why it gets printed like that, but I think the condition you want is:

```coffee
'.type == "counter" || .type == "gauge"'
```

Ref: https://vector.dev/docs/reference/configuration/transforms/remap/#event-data-model

**Reply by Quyzi** [reply]
Ah okay. I didn't see that. I read through https://vector.dev/docs/about/under-the-hood/architecture/data-model/metric/ 
Thank you!

**Reply by spencergilbert** [reply]
It's definitely in an awkward place to find - I'm hoping we'll have it in a more natural place after we release our site redesign.
